package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -329,20 +247,26 @@ cdef class NSResults(object):
     cdef np.ndarray pair_coordinates_buffer
 
     def __init__(self, real cutoff, real[:, ::1]coords, ns_int[:] search_ids, debug=False):
-
         self.debug = debug
         self.cutoff = cutoff
         self.coords = coords
         self.search_ids = search_ids
 
         # Preallocate memory
         self.allocation_size = search_ids.shape[0] + 1
-        self.pairs = <ipair *> PyMem_Malloc(sizeof(ipair) * self.allocation_size)
-        if not self.pairs:
-            MemoryError("Could not allocate memory for NSResults.pairs ({} bits requested)".format(sizeof(ipair) * self.allocation_size))
-        self.pair_distances2 = <real *> PyMem_Malloc(sizeof(real) * self.allocation_size)
-        if not self.pair_distances2:
-            raise MemoryError("Could not allocate memory for NSResults.pair_distances2 ({} bits requested)".format(sizeof(real) * self.allocation_size))
+        if not self.pairs and not self.pair_distances2:

```

Comment:
if they are not allocated we use the old code. In case `__init__` was already called we reallocate them. This should allow `__init__` to be called twice, even by a user manually.